src/SignalR/common/Http.Connections/src/Internal/HttpConnectionsMetrics.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.Extensions.Metrics;
+
+namespace Microsoft.AspNetCore.Http.Connections.Internal;
+
+internal sealed class HttpConnectionsMetrics : IDisposable
+{
+    public const string MeterName = "Microsoft.AspNetCore.Http.Connections";
+
+    private readonly Meter _meter;
+    private readonly UpDownCounter<long> _currentConnectionsCounter;
+    private readonly Histogram<double> _connectionDuration;
+
+    public HttpConnectionsMetrics(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.CreateMeter(MeterName);
+
+        _currentConnectionsCounter = _meter.CreateUpDownCounter<long>(
+            "current-connections",
+            description: "Number of concurrent connections that are currently active on the server.");
+
+        _connectionDuration = _meter.CreateHistogram<double>(
+            "connection-duration",
+            unit: "s",
+            description: "The duration of connections on the server.");
+    }
+
+    public void ConnectionStart()
+    {
+        _currentConnectionsCounter.Add(1);
```

Comment:
Would it be interesting to include the transport type as a tag? e.g. LongPolling, WebSocket

Reply:
Unfortunately, the transport type is negotiated after this point. Can't add it to this up/down counter.

However, I'll add the transport type to the connection duration counter.